shared/src/main/kotlin/com/egm/stellio/shared/model/ApiExceptions.kt

```diff
@@ -2,24 +2,129 @@ package com.egm.stellio.shared.model
 
 import com.apicatalog.jsonld.JsonLdError
 import com.apicatalog.jsonld.JsonLdErrorCode
+import com.fasterxml.jackson.annotation.JsonIgnore
+import com.fasterxml.jackson.annotation.JsonProperty
+import org.slf4j.LoggerFactory
+import org.springframework.http.HttpStatus
+import org.springframework.http.MediaType
+import org.springframework.http.ProblemDetail
+import org.springframework.http.ResponseEntity
+import java.net.URI
+
+const val DEFAULT_DETAIL = "If you have difficulty identifying the issue," +
+    "you can check common problem on https://stellio.readthedocs.io/en/latest/TROUBLESHOOT.html" +
+    "If you believe this is a bug, you can report it by creating an issue on" +
+    "https://github.com/stellio-hub/stellio-context-broker"
 
 sealed class APIException(
-    override val message: String
-) : Exception(message)
-
-data class InvalidRequestException(override val message: String) : APIException(message)
-data class BadRequestDataException(override val message: String) : APIException(message)
-data class AlreadyExistsException(override val message: String) : APIException(message)
-data class OperationNotSupportedException(override val message: String) : APIException(message)
-data class ResourceNotFoundException(override val message: String) : APIException(message)
-data class InternalErrorException(override val message: String) : APIException(message)
-data class TooComplexQueryException(override val message: String) : APIException(message)
-data class TooManyResultsException(override val message: String) : APIException(message)
-data class AccessDeniedException(override val message: String) : APIException(message)
-data class NotImplementedException(override val message: String) : APIException(message)
-data class LdContextNotAvailableException(override val message: String) : APIException(message)
-data class NonexistentTenantException(override val message: String) : APIException(message)
-data class NotAcceptableException(override val message: String) : APIException(message)
+    val type: URI,
+    @JsonIgnore
+    val status: HttpStatus,
+    @JsonProperty("detail")
+    override val message: String,
+    open val detail: String = DEFAULT_DETAIL
```

Comment:
the value of the `JsonProperty` is kind of conflicting with the `detail` member.

Reply:
don't think the Jackson annotations are even used since it is always serialized as a `ProblemDetails` structure?